src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java

```diff
@@ -1615,14 +1645,16 @@ private V awaitConversionResultAndRetryToFindVariant(B blob, String variantName,
     protected abstract boolean markConversionAttempt(V variant) throws Exception;
 
     /**
-     * Spawns a thread which will actually invoke the appropriate conversion pipeline.
+     * Spawns a thread which will actually invoke the appropriate conversion pipeline asynchronously.
      *
      * @param blob    the blob for which the variant is to be created
      * @param variant the variant to generate
+     * @return a future holding the conversion process
      */
-    private void invokeConversionPipelineAsync(B blob, V variant) {
+    private Future invokeConversionPipelineAsync(B blob, V variant) {
```

Comment:
Thinking loud: aren't we making it way too complex? The VideoProcessingJob could call the next 2 lines directly, and handle the Future on its own, leaving stuff like max duration at job level. Even logging, since you than have access to the Future's onFailureCallback and the like.